qa/tasks/mgr/dashboard/test_health.py

```diff
@@ -227,8 +227,8 @@ def test_full_health(self):
                     'gid': int,
                     'name': str,
                     'mgr_features': int
-                }))
-            }),
+                }, allow_unknown=True))
+            }, allow_unknown=True),
```

Comment:
Just grepped that directory and `allow_unknown=True` is used 47 times (out of >100 `JObj` instances, so ~50%). @votdev @rjfd @sebastian-philipp @ricardoasmarques : Would it make sense to change the default behaviour  of `JObj` from `False` to `True`?

My point is: are `unknown_keys` catching real issues or just triggering QA failures for (harmless) additions to the core data structures? If the reason is that these additions are breaking the strong-typed structs in the front-end, perhaps we should rely on interfaces instead of types, or validating and stopping those changes at the back-end, instead of forwarding them to the end-user and performing the validation there?

Reply:
`unknown_keys=True` by default is a real problem, if they originate from `mgr/dashboard`, because then we don't know what we're returning to the frontend.

Reply:
> `unknown_keys=True` by default is a real problem, if unknown keys originate from `mgr/dashboard`, because then we don't know what we're returning to the frontend.

But... that's what's happening right now. Our back-end is basically doing a `SELECT * FROM <some_Ceph_map>` and passing that dump to the front-end. IMHO, instead of failing when Ceph core returns some extra field we didn't expect, we should explicitly compose API payloads with the data we have committed to deliver.

Nevertheless, if we are returning 1:1 mappings to Ceph structs, we perhaps shouldn't be that picky and allow unknown fields hitting the front-end. That means we just expect the incoming data to meet an interface, but not to strictly fullfill all of it. I like the pragmatic approach we've taken so far, but it seems that we daren't take it all the way :sweat_smile:

Reply:
> > `unknown_keys=True` by default is a real problem, if unknown keys originate from `mgr/dashboard`, because then we don't know what we're returning to the frontend.
> 
> But... that's what's happening right now. Our back-end is basically doing a `SELECT * FROM <some_Ceph_map>` and passing that dump to the front-end.

yes, in all of these cases, I'm :+1: for `unknown_keys=True`

> IMHO, instead of failing when Ceph core returns some extra field we didn't expect, we should explicitly compose API payloads with the data we have committed to deliver.

Yeah, we've done that already in openATTIC. Was a big mess. I wouldn't do it again.